include/lsst/ctrl/events/Transmitter.h

```diff
@@ -0,0 +1,116 @@
+// -*- lsst-c++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/**
+ * @file Transmitter.h
+ *
+ * @ingroup ctrl/events
+ *
+ * @brief defines the Transmitter class
+ */
+
+#ifndef LSST_CTRL_EVENTS_TRANSMITTER_H
+#define LSST_CTRL_EVENTS_TRANSMITTER_H
+
+#include <cms/Connection.h>
+#include <cms/Session.h>
+#include <cms/TextMessage.h>
+
+#include <stdlib.h>
+#include <iostream>
+
+#include "lsst/daf/base/PropertySet.h"
+
+#include "lsst/ctrl/events/Event.h"
+#include "lsst/ctrl/events/EventBroker.h"
+
+using lsst::daf::base::PropertySet;
+
+namespace lsst {
+namespace ctrl {
+namespace events {
+
+/**
+ * @class Transmitter
+ * @brief Transmit events to the event bus
+ */
+
+class Transmitter
+{
+public:
+
+    Transmitter();
+
+
+    /**
+     * @brief destructor
+     */
+    virtual ~Transmitter();
+
+    /**
+     * @brief Publish an Event to this object's topic
+     * @param event an Event to publish
+     */
+    void publishEvent(Event& event);
+
+    /**
+     * @brief get the destination property name
+     * @note This is the TYPE of the destination we're using, either a TOPIC or a QUEUE
+     */
+    virtual std::string getDestinationPropertyName() { return NULL; }
```

Comment:
Sorry, I realize it's too late for comments, but I still have to say it :smile: 
I hope this method is never called (overridden in all subclasses), constructing an `std::string` from zero pointer will very likely result in a crash. If you want an empty string use `""` or explicit `std::string()` (I prefer latter).